### ISSUE TITLE
Ballerina shell sometimes does not handle newline in double quotes properly

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-lang#28313]
In the shell, if I type:

```
"abc
```

the shell rightly gives me an error immediately

```
| Tree parsing failed: missing double quote
| "abc;
| ^
| Parsing aborted because of errors.
```

(although the position is wrong).

However, if I do:

```
io:println("abc
```

it waits for the next line, instead of giving an error right away.

**Comment by kdsuneraavinash** [comment]
That behavior is because shell is waiting for the paren to be closed, not the quotes.
But this is still an issue, because quotes cannot be continued on new lines regardless.

I think we might have to implement a proper way to detect if the code is partially entered. 
This, #28282 and #28274 can also be fixed by that.

**Comment by kdsuneraavinash** [comment]
Fixed by #28344 and #28321
